java/arcs/android/common/resurrection/DbHelper.kt

```diff
@@ -94,50 +96,60 @@ class DbHelper(
                 "component_class",
                 request.componentName.className
             )
+            notifierValues.put(
+                "notifier_id",
+                request.notifierId
+            )
             notifierValues.put("notification_key", it.toString())
             insert("requested_notifiers", null, notifierValues)
         }
     }
 
     /** Unregisters a [component] for resurrection. */
-    fun unregisterRequest(component: ComponentName) = writableDatabase.transaction {
-        val componentArgs = arrayOf(component.packageName, component.className)
+    fun unregisterRequest(
+        component: ComponentName,
+        notifierId: String
+    ) = writableDatabase.transaction {
+        val deletionArgs = arrayOf(component.packageName, component.className, notifierId)
         execSQL(
             """
                 DELETE FROM requested_notifiers 
-                WHERE component_package = ? AND component_class = ?
+                WHERE component_package = ? AND component_class = ? AND notifier_id = ?
             """,
-            componentArgs
+            deletionArgs
         )
         execSQL(
             """
                 DELETE FROM resurrection_requests
-                WHERE component_package = ? AND component_class = ?
+                WHERE component_package = ? AND component_class = ? AND notifier_id = ?
             """,
-            componentArgs
+            deletionArgs
         )
     }
 
     /**
      * Gets all registered [ResurrectionRequest]s from the database.
      */
     fun getRegistrations(): List<ResurrectionRequest> {
-        val notifiersByComponentName = mutableMapOf<ComponentName, MutableList<StorageKey>>()
+        val notifiersByComponentName =
+            mutableMapOf<Pair<ComponentName, String>, MutableList<StorageKey>>()
```

Comment:
I feel like it'd be cleaner to create a little purpose-built data class here, instead of using Pair.